### PR TITLE
Update darwin framework tool project to mimic gn compile flags for KVS

### PR DESCRIPTION
**Pull request by krypton36**
- Update compile flags to mirror GN build of DFT.

### CODE REVIEW

File: src/darwin/Framework/Matter.xcodeproj/project.pbxproj
Comment (bzbarsky-apple) [inline src/darwin/Framework/Matter.xcodeproj/project.pbxproj:1]: Why is this needed?  Shouldn't darwin-framework-tool just be compiled with `CHIP_DISABLE_PLATFORM_KVS` set?

File: src/darwin/Framework/Matter.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1813,9 +1813,9 @@
 					CHIP_HAVE_CONFIG_H,
 					"CHIP_ADDRESS_RESOLVE_IMPL_INCLUDE_HEADER=<lib/address_resolve/AddressResolve_DefaultImpl.h>",
 					CONFIG_BUILD_FOR_HOST_UNIT_TEST,
-					CONFIG_USE_LOCAL_STORAGE,
 					"CHIP_CONFIG_SKIP_APP_SPECIFIC_GENERATED_HEADER_INCLUDES=1",
 					"CONFIG_USE_INTERACTIVE_MODE=1",
+					CHIP_DISABLE_PLATFORM_KVS,
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/Matter.xcodeproj/project.pbxproj:1818]:
```suggestion
					"CHIP_DISABLE_PLATFORM_KVS=1",
```
no?

File: src/darwin/Framework/Matter.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1883,9 +1883,9 @@
...
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/Matter.xcodeproj/project.pbxproj:1888]:
```suggestion
					"CHIP_DISABLE_PLATFORM_KVS=1",
```

File: src/darwin/Framework/Matter.xcodeproj/project.pbxproj
Comment (bzbarsky-apple) [inline src/darwin/Framework/Matter.xcodeproj/project.pbxproj:1]: The commit message (and PR description) needs to be fixed to reflect what's actually changing.

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform-specific change with platform owner review.

**Review by andy31415 — APPROVED** [review]
